runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/ProcessImportDialog.utils.tsx

```diff
@@ -0,0 +1,70 @@
+import { List, ListItem, Typography } from '@mui/material';
+import { Credential } from 'runbotics-common';
+
+import { translate } from '#src-app/hooks/useTranslations';
+import { credentialsActions } from '#src-app/store/slices/Credentials';
+
+import { AdditionalInfo } from './Modeler/BpmnModeler';
+
+interface ClearImportedCredentialsParams {
+    definition: string;
+    additionalInfo: AdditionalInfo;
+    setOpen: React.Dispatch<React.SetStateAction<boolean>>;
+    importDraft: (definition: string, additionalInfo: AdditionalInfo) => void;
+}
+
+export const mapCredentialsToList = (credentials: Credential[]) => (
+    <List>
+        {credentials && credentials.length && [...new Map(credentials.map(c => [c.id, c])).values()]
```

Comment:
You could do shorter - credentials?.length

Reply:
Also, is it necessary to make new Map and convert it instantly to array instead of returning just array?

Reply:
it's removing duplicates by creating object of { <credId>: <credential> } (and because the object cannot have two the same keys it works)

Reply:
So, I'm keeping it as it is